lib/web_ui/test/canvaskit/embedded_views_diff_test.dart

```diff
@@ -0,0 +1,134 @@
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[6]);
+      expect(result.viewsToRemove, <int>[]);
+      expect(result.addToBeginning, isFalse);
+
+      result = diffViewList(<int>[3, 4, 5, 6], <int>[3, 4, 5]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[]);
+      expect(result.viewsToRemove, <int>[6]);
+
+      result = diffViewList(<int>[3, 4, 5, 6], <int>[4, 5, 6]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[]);
+      expect(result.viewsToRemove, <int>[3]);
+      expect(result.addToBeginning, isFalse);
+
+      result = diffViewList(<int>[3, 4, 5, 6, 7, 8], <int>[3, 4, 5]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[]);
+      expect(result.viewsToRemove, <int>[6, 7, 8]);
+
+      result = diffViewList(<int>[1, 2, 3, 4, 5, 6], <int>[4, 5, 6]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[]);
+      expect(result.viewsToRemove, <int>[1, 2, 3]);
+      expect(result.addToBeginning, isFalse);
+
+      result = diffViewList(<int>[3, 4, 5, 6, 7, 8], <int>[2, 3, 4, 5]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[2]);
+      expect(result.viewsToRemove, <int>[6, 7, 8]);
+      expect(result.addToBeginning, isTrue);
+      expect(result.viewToInsertBefore, 3);
+
+      result = diffViewList(<int>[1, 2, 3, 4, 5, 6], <int>[4, 5, 6, 7]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[7]);
+      expect(result.viewsToRemove, <int>[1, 2, 3]);
+      expect(result.addToBeginning, isFalse);
+
+      result = diffViewList(<int>[1, 2, 3], <int>[4, 5]);
+      expect(result, isNull);
+
+      result = diffViewList(<int>[1, 2, 3, 4], <int>[2, 3, 5, 4]);
+      expect(result, isNull);
+
+      result = diffViewList(<int>[3, 4], <int>[1, 2, 3, 4, 5, 6]);
+      expect(result, isNull);
+
+      result = diffViewList(<int>[1, 2, 3, 4, 5], <int>[2, 3, 4]);
+      expect(result, isNull);
+    });
+
+    test('works for flutter/flutter#101580', () {
+      ViewListDiffResult? result;
+
+      // Reverse the list
+      result = diffViewList(<int>[1, 2, 3, 4], <int>[4, 3, 2, 1]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[3, 2, 1]);
+      expect(result.viewsToRemove, <int>[1, 2, 3]);
+      expect(result.addToBeginning, isFalse);
+
+      // Sort the list
+      result = diffViewList(<int>[3, 4, 1, 2], <int>[1, 2, 3, 4]);
+      expect(result, isNotNull);
+      expect(result!.viewsToAdd, <int>[3, 4]);
+      expect(result.viewsToRemove, <int>[3, 4]);
+      expect(result.addToBeginning, isFalse);
+
+      // Move last view to the beginning
+      // (The algo explores the diff from left to right, but in this case, it'd
+      // more efficient to add [1] at the beginning. Maybe we should compute both
+      // diffs, from left and right, and return the one that results in fewer
+      // add/remove operations?)
```

Comment:
I think this is exactly what the algo is doing now, so this comment is a lie :P

Reply:
done